File: backend/src/LinearPrecision.Api/Entities/TimeEntry.cs
using LinearPrecision.Shared.Domain;

namespace LinearPrecision.Api.Entities;

public class TimeEntry : TenantEntity, IAuditable
{
    public Guid UserId { get; set; }
    public User User { get; set; } = null!;
    public Guid? TaskId { get; set; }
    public TaskItem? Task { get; set; }
    public Guid? ProjectId { get; set; }
    public Project? Project { get; set; }
    public string? Description { get; set; }
    public DateTime StartTime { get; set; }
    public DateTime? EndTime { get; set; }
    public int DurationMinutes { get; set; }
    public bool IsBillable { get; set; }
    public decimal? HourlyRate { get; set; }

    // IAuditable
    public Guid? CreatedBy { get; set; }
    public Guid? UpdatedBy { get; set; }
}

File: lib/api/contracts/time-entry.ts
export interface TimeEntryUserBrief {
  id: string;
  fullName: string;
  avatarUrl?: string | null;
}

export interface TimeEntryResponse {
  id: string;
  user: TimeEntryUserBrief;
  taskId?: string | null;
  projectId?: string | null;
  description?: string | null;
  startTime: string;
  endTime?: string | null;
  durationMinutes: number;
  isBillable: boolean;
  hourlyRate?: number | null;
  createdAt: string;
  updatedAt: string;
}

export interface CreateTimeEntryRequest {
  taskId?: string | null;
  projectId?: string | null;
  description?: string | null;
  startTime: string;
  endTime?: string | null;
  durationMinutes?: number | null;
  isBillable?: boolean | null;
  hourlyRate?: number | null;
}

export interface UpdateTimeEntryRequest {
  taskId?: string | null;
  projectId?: string | null;
  description?: string | null;
  startTime: string;
  endTime?: string | null;
  durationMinutes?: number | null;
  isBillable?: boolean | null;
  hourlyRate?: number | null;
}

export interface StartTimerRequest {
  taskId?: string | null;
  projectId?: string | null;
  description?: string | null;
  isBillable?: boolean | null;
}
